packages/firestore/src/lite-api/query.ts

```diff
@@ -193,6 +201,92 @@ export function where(
   return new QueryFilterConstraint(field, op, value);
 }
 
+class QueryCompositeFilterConstraint extends QueryConstraint {
+  constructor(
+    readonly type: 'or' | 'and',
```

Comment:
can we use `CompositeOperator` ?

Reply:
In QueryConstraint, `type` is defined as `abstract readonly type: QueryConstraintType;`.

It looks like I could use `CompositeOperator`, but I don't think mixing the types there is a best practice. Further, the current code is consistent with other QueryConstraint subtypes, such as [QueryLimitConstraint](https://github.com/firebase/firebase-js-sdk/blob/7cad614ec2d2a34b40a3c24443c4f35571e3e68c/packages/firestore/src/lite-api/query.ts#L283).

Reply:
Ah, yes. Good call.